### PR TITLE
Remove unneeded type SecureStringHasher

### PR DESCRIPTION
**Pull request by GrabYourPitchforks**
.NET Core always enables `string.GetHashCode` randomization. Since `System.Private.Xml` is built only for .NET Core (not downlevel platforms) there's no need for it to carry its own `SecureStringHasher` type.

### CODE REVIEW

File: src/libraries/System.Private.Xml/src/System/Xml/XmlQualifiedName.cs [hunk 1]
```diff
@@ -2,7 +2,9 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-// Warning: Do not port implementation of GetHashCode to Desktop. See notes in SecureStringHasher
+// Warning: Do not port implementation of GetHashCode to Desktop.
```
Comment (stephentoub) [inline src/libraries/System.Private.Xml/src/System/Xml/XmlQualifiedName.cs]:
```suggestion
// Warning: Do not port implementation of GetHashCode to .NET Framework.
```

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Private.Xml/src/System/Xml/XmlQualifiedName.cs (hunk 1 above)
Comment (jkotas) [reply src/libraries/System.Private.Xml/src/System/Xml/XmlQualifiedName.cs]: This line that refers to .NET Framework can be deleted. The risk that somebody would accidentally port this to .NET Framework is exactly zero.

Comment (stephentoub) [reply src/libraries/System.Private.Xml/src/System/Xml/XmlQualifiedName.cs]: > exactly zero

:-)

**Review by buyaa-n — APPROVED** [review]
Thanks

File: src/libraries/System.Private.Xml/src/System/Xml/XmlQualifiedName.cs (hunk 1 above)
Comment (GrabYourPitchforks) [reply src/libraries/System.Private.Xml/src/System/Xml/XmlQualifiedName.cs]: I never knew Jan to be such an optimist. :)

File: src/libraries/System.Private.Xml/src/System/Xml/BinaryXml/XmlBinaryReader.cs [hunk 2]
```diff
@@ -98,9 +98,10 @@ public override int GetHashCode()
                 return this.prefix.GetHashCode() ^ this.localname.GetHashCode();
             }
 
-            public int GetNSHashCode(SecureStringHasher hasher)
+            public int GetNSHashCode()
             {
-                return hasher.GetHashCode(this.namespaceUri) ^ hasher.GetHashCode(this.localname);
+                // ValueTuple uses HashCode.Combine with the randomized string.GetHashCode as inputs
```
Comment (jkotas) [inline src/libraries/System.Private.Xml/src/System/Xml/BinaryXml/XmlBinaryReader.cs]: Would it be better to do `HashCode.Combine(namespaceUri, localname)` directly here? Avoids layers of indirection.

**Review by jkotas — APPROVED** [review]

File: src/libraries/System.Private.Xml/src/System/Xml/BinaryXml/XmlBinaryReader.cs (hunk 2 above)
Comment (jkotas) [reply src/libraries/System.Private.Xml/src/System/Xml/BinaryXml/XmlBinaryReader.cs]: #31784

**Pull request merged**